geomstats/geometry/hypersphere.py

```diff
@@ -300,12 +307,13 @@ def random_uniform(self, n_samples=1):
         samples = gs.random.normal(size=size)
         while True:
             norms = gs.linalg.norm(samples, axis=1)
-            indcs = gs.isclose(norms, 0.0)
+            indcs = gs.isclose(norms, 0.0, atol=tol)
             num_bad_samples = gs.sum(indcs)
             if num_bad_samples == 0:
                 break
-            samples[indcs, :] = gs.random.normal(
+            new_samples = gs.random.normal(
                 size=(num_bad_samples, self.dim + 1))
+            samples = self._replace_values(samples, new_samples, indcs)
```

Comment:
I came up with an idea:
`samples = gs.concatenate([samples[~indcs], new_samples])`
what do you think?

Reply:
That's smart, since we don't care about the order !

However, to have that part of code tested, we either have to do "seed mining", or define it in an auxiliary function. Any idea of how it could be named ? (replace_values would not be completely appropriate anymore)

Reply:
True, maybe `_enforce_non_zeros`

Reply:
It does not explictely deal with the zeros, simply the replacement. Maybe _replace_unordered_values ?
I don't know whether it could be interesting to have both (ordered and unordered alternatives).